pico/tools/src/main/java/io/helidon/pico/tools/DefaultApplicationCreator.java

```diff
@@ -348,7 +349,7 @@ List<String> toInjectionPlanBindings(ServiceProvider<?> sp) {
                     }
                     line.append(", ").append(((Class<?>) target).getName()).append(".class");
                 } else if (ipQualified.isEmpty()) {
-                    // no-op
+                    assert (true); // left here for breakpoint debugging purposes
```

Comment:
remove this line, and collapse into the next `else if`. This branch is useless.
And you still can have a breakpoint here (use condition breakpoint with `isQualified.isEmpty()`)

Reply:
See previous commets.

Reply:
pushed